rclpy/rclpy/impl/rcutils_logger.py

```diff
@@ -221,15 +220,30 @@ def __init__(self, name=''):
         self.name = name
         self.contexts = {}
 
+    def get_child(self, name):
+        return RcutilsLogger(name=self.name + '.' + name)
```

Comment:
Does this need to handle the case where `self.name` is an empty string?

Reply:
Checks added in 52a36611bbf91c92f4a79623b55b83e59b07344c

Reply:
I was aiming for `self.name` being empty which results in the new loggers name to be ".<name>".

Reply:
It doesn't _need_ to but we could; the rcutils logger hierarchy will still be functional in the case of empty name tokens. I was taking the approach of allowing `''.child_name` to be a valid logger name if a user wants to do so (since they're calling `get_child` as opposed to `get_named_logger`).

If instead we remove the prefixing dot then we have returned an independent sibling instead of a child; in your opinion is that the expected behaviour for the edge case, or was your question just if it would lead to undefined behaviour?

Reply:
The question is how is the topology of the loggers? Does `foo` and `bar` not share a common parent in your case? If not, why not? If yes, what would that be if not ``?

I though of the loggers as a hierarchy in the form of a tree with a single root. `` being the root of it and `"anything_without_a_dot"` are the direct children of it.

Reply:
Ah, I see where the difference in understanding has come from now: in my mind there didn't need to be a single root and you could create a logger that is not attached to the root if you wanted to.

The idea of a single root matches how log4cxx and python logging module do it, so I'll switch to that. Every logger will at some point be attached to the root logger, but the root logger won't have a name. And then, yes, the dot prefix will be removed when appending a direct child to the root logger.

I'll update rcutils to support this (there isn't really a notion of a root logger in https://github.com/ros2/rcutils/pull/57, but "default" settings).

Reply:
Does this better match what you had in mind? 21e5c4438ce28cc004f3572d83173597b6efbb1c
Now a call to `get_named_logger` just calls `get_child` on the root logger: all loggers are descendants of the root logger.

4c8bdbae96007e481ced2cf014cc74abcab19878 removes the notion of default configuration in favour of configuring the root logger.

Reply:
:+1: looks good to me.